src/main/main.go

```diff
@@ -0,0 +1,367 @@
+		time.Sleep(5 * time.Second)
+
+		if timeused >= limit {
+			return fmt.Errorf("time limit achieved waiting for job to finish")
+		}
+		timeused += 5
+	}
+}
+
+func teardown(ctx context.Context, pubCli, privCli *skupper_ocp_smoke.Client) error {
+
+	step := "teardown"
+	log.Printf("%s : Starting Teardown", step)
+
+	cliList := []*skupper_ocp_smoke.Client{
+		pubCli,
+		privCli,
+	}
+
+	// Delete the subscription
+	for _, cli := range cliList {
+		err := cli.DeleteSubscription("teardown")
+		if err != nil {
+			return fmt.Errorf("%s : %s", step, err.Error())
+		}
+		skupper_ocp_smoke.PrintIfDebug(step, ": subscription deleted from namespace", cli.Namespace)
+	}
+
+	// Delete the operator group
+	for _, cli := range cliList {
+		err := cli.DeleteOperatorGroup("teardown")
+		if err != nil {
+			return fmt.Errorf("%s : %s", step, err.Error())
+		}
+		skupper_ocp_smoke.PrintIfDebug(step, ": OperatorGroup deleted from", cli.Namespace)
+	}
+
+	for _, cli := range cliList {
+		if err := cli.DeleteNamespace(ctx, "teardown"); err != nil {
+			return fmt.Errorf("%s : %s", step, err.Error())
+		}
+		skupper_ocp_smoke.PrintIfDebug(step, ": Namespace deletion scheduled for", cli.Namespace)
+
+		if err := cli.WaitForNSDeletion(ctx, cli.WaitLimit(), "teardown"); err != nil {
+			return fmt.Errorf("%s : %s", step, err.Error())
+		}
+	}
+	log.Printf("%s : Teardown finished", step)
+	return nil
+}
+
+func main() {
+	err := run()
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+
+	log.Printf("main : Starting main test")
+
+	skupper_ocp_smoke.PrintIfDebug("main : Debug detail")
+
+	if skupper_ocp_smoke.Startingcsv != "" {
+		skupper_ocp_smoke.PrintIfDebug("main : Using specific startingCSV = ", skupper_ocp_smoke.Startingcsv)
+	}
+
+	if skupper_ocp_smoke.Waitlimit != skupper_ocp_smoke.WAITLIMIT {
+		skupper_ocp_smoke.PrintIfDebug("main : Using specific timeout value = ", strconv.Itoa(skupper_ocp_smoke.Waitlimit))
+	}
+
+	log.Printf("main : Get Kubeconfig settings\n")
+	pubKubeconfigFile := skupper_ocp_smoke.KubeConfigDefault(skupper_ocp_smoke.PUBKUBECONFIGFILE, "PUBKUBECONFIG")
+	if pubKubeconfigFile == "" {
+		return fmt.Errorf("unable to determine the kubeconfig for Public instance. Aborting")
+	}
+
+	privKubeconfigFile := skupper_ocp_smoke.KubeConfigDefault(skupper_ocp_smoke.PRIVKUBECONFIGFILE, "PRIVKUBECONFIG")
+	if privKubeconfigFile == "" {
+		return fmt.Errorf("unable to determine the kubeconfig for Private instance. Aborting")
+	}
+
+	skupper_ocp_smoke.PrintIfDebug("main : Kubeconfig for public namespace = ", os.Getenv("PUBKUBECONFIG"))
+	skupper_ocp_smoke.PrintIfDebug("main : Kubeconfig for private namespace = ", os.Getenv("PRIVKUBECONFIG"))
+
+	// Context and CLI
+	ctx, cn := context.WithTimeout(context.Background(), time.Minute*30)
+	defer cn()
+
+	pubCli, _ := skupper_ocp_smoke.NewClient("", pubKubeconfigFile, skupper_ocp_smoke.PUBNS)
+	privCli, _ := skupper_ocp_smoke.NewClient("", privKubeconfigFile, skupper_ocp_smoke.PRIVNS)
```

Comment:
The `NewClient` function is returning a nil value to me.
It is better to check/log eventual errors returned here.

Reply:
Added an error checking